### DISCUSSION TITLE
Transposition operators

### DISCUSSION THREAD

**Discussion by daveyarwood** [alda-lang/alda#428] [Ideas]
Moved from https://github.com/alda-lang/alda/issues/129.

See that issue for details / discussion.

**Comment by daveyarwood** [comment]
We should consider whether this should be implemented as alda.lisp (inline Clojure code) functions rather than as a new operator. This is an overall design goal of mine.

**Comment by elydpg** [comment]
Transposition is going to be complicated. Unlike attributes like tempo, it may be desirable to add and subtract the values from one another upon their setting. Suppose a composer writes a piano part in concert pitch and a trumpet part in B♭, then decides that later they want to use the entire thing again, only both parts are to be played a semitone higher. 

Personally, I always write in concert pitch, but the transpose function would be useful for modulating preexisting parts. I may also desire to play entire scores in different keys.

I'm also thinking we should combine this issue and alda-lang/alda-core#21, as they are very similar in nature.
